zaza/openstack/charm_tests/vault/utils.py

```diff
@@ -280,6 +281,65 @@ def ensure_secret_backend(client):
         pass
 
 
+def wait_until_ready(clients):
+    """Wait until vault is ready (for all clients).
+
+    Raise an error if it's not ready within a reasonable time.
+
+    :param clients: Clients to use to talk to vault
+    :type clients: List[CharmVaultClient]
+    :raises: AssertionError
+    """
+    for client in clients:
+        for attempt in tenacity.Retrying(
+            reraise=True,
+            wait=tenacity.wait_fixed(10),
+            stop=tenacity.stop_after_attempt(12),  # wait for max 2 minutes
+        ):
+            with attempt:
+                ha_status = client.hvac_client.ha_status
+                if (
+                    not ha_status.get('leader_address') and
+                    ha_status.get('ha_enabled')
```

Comment:
Am I reading this correctly. "raise an exception if it's not got a leader address AND does have ha_enabled"?  So this function only works for units that are 'ha_enabled'?  That might be useful for the docstring and the name of the function?